src/pages/blog-page/index.tsx

```diff
@@ -121,21 +127,66 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const BlogPage = () => {
+const BlogPage: FC<{ data: BlogGQL }> = ({ data }) => {
+  const blogData = data.markdownRemark.blogPage;
   const classes = useStyles();
+  const { componentType, isDesktop } = useComponentType();
+
+  const [selectedBlogpost, setSelectedBlogpost] = useState(-1);
+
+  // no blogpost will have index equal to -1 therefore no blogpost will be selected
+  const handleCloseBlogpost = () => {
+    setSelectedBlogpost(-1);
+  };
+  // if is first blogpost, choose last blogpost
+  const handlePrevBlogpost = (index: number) => {
+    setSelectedBlogpost(index === 0 ? blogData.blog_post.length - 1 : index - 1);
+  };
+
+  // if is last blogpost, choose first blogpost
+  const handleNextBlogpost = (index: number) => {
+    setSelectedBlogpost(index === blogData.blog_post.length - 1 ? 0 : index + 1);
+  };
+
   return (
     <Container className={classes.container}>
-      <Box className={classes.aside}>
-        <DetailsCard type="desktop" {...userData}></DetailsCard>
-      </Box>
+      {isDesktop && (
+        <Box className={classes.aside}>
+          <DetailsCard type={componentType} />
+        </Box>
+      )}
       <Box className={classes.main}>
         <Box className={classes.navbar}>NavBar</Box>
         <Box className={classes.mainContent}>
           <Box className={classes.blogContainer}>
             <SectionTitle className={classes.title}>Blog</SectionTitle>
             <Box className={classes.blogPosts}>
-              {[...Array(4)].map((_, index) => (
-                <BlogPost key={index} className={classes.blogPost} {...blogData} />
+              {blogData.blog_post.map((blogPost, index) => (
+                <div key={index}>
```

Comment:
Don't use the index as key

Reply:
Good catch, will replace in portfolio page aswell